service/controller/app/resource/chartoperator/delete.go

```diff
@@ -22,11 +25,35 @@ func (r Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 	// Resource is used to bootstrap chart-operator. So for other apps we can
 	// skip this step.
 	if key.AppName(cr) != key.ChartOperatorAppName {
-		r.logger.Debugf(ctx, "no need to install chart-operator for %#q", key.AppName(cr))
-		r.logger.Debugf(ctx, "canceling resource")
 		return nil
 	}
 
+	// Check if cluster is being deleted
+	clusterID := key.ClusterID(cr)
+
+	if clusterID != "" {
+		capiCluster := &capi.Cluster{}
+		err = r.ctrlClient.Get(
+			ctx,
+			types.NamespacedName{Name: clusterID, Namespace: cr.Namespace},
+			capiCluster,
+		)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if capiCluster.GetDeletionTimestamp() != nil {
+			// Canceling reconciliation stops processing of further
+			// resource and makes the OperatorKit to remove the finalizer.
+			// This is what we want for Chart Operator upon cluster deletion.
+			r.logger.Debugf(ctx, "workload cluster is being deleted, no need to try to remove the chart-operator")
+			r.logger.Debugf(ctx, "canceling reconciliation")
+
+			reconciliationcanceledcontext.SetCanceled(ctx)
```

Comment:
I did not dig in deep into this, just a thought: should we not remove the finalizer in this condition and then cancel? Otherwise wont the WC chart op app cr stick around forever? 🤔